### ISSUE TITLE
Hints with same name

### ISSUE THREAD

**Issue by georgefst** [ndmitchell/hlint#1164]
The hints triggered by:

```hs
f <$> asks g
```

and:

```hs
f =<< g <$> x
```

seem, to me, to be about two quite different things. Different enough, in fact, that I find myself wanting to disable the former and not the latter. But since both have the same name, `"Redundant <$>"`, this doesn't seem to be possible. Perhaps one could be renamed? Obviously, that does bring some backwards compatibility concerns...

**Comment by ndmitchell** [comment]
Not too fussed about the backwards compatibility concerns - we add new hints all the time, so people should expect these to appear semi regularly. Hints by default have an auto-generated name, unless its overridden in the .hlint.yaml. I agree the first should definitely have a different name, and the second seems a bit wrong too. Want to send a diff? I'm no good with naming :)
